R/plotconfiguration-histogram.R

```diff
@@ -0,0 +1,142 @@
+#' @title HistogramPlotConfiguration
+#' @docType class
+#' @description  Plot Configuration for Histograms
+#' @export
+HistogramPlotConfiguration <- R6::R6Class(
+  "HistogramPlotConfiguration",
+  inherit = PlotConfiguration,
+
+  public = list(
+    mapData = NULL,
+    bins = NULL,
+    binWidth = NULL,
+    verticalLineFunctions = NULL,
+    verticalLineFunctionNames = NULL,
+
+    #verticalLineProperties = NULL,
+
+    initialize = function(#verticalLineProperties = tlfEnv$currentTheme$verticalLineProperties,
+      title = "Histogram",
+      subtitle = paste("Date:", format(Sys.Date(), "%y-%m-%d")),
+      xlabel = NULL,
+      ylabel = NULL,
+      watermark = tlfEnv$currentTheme$watermarkText,
+      data = NULL,
+      metaData = NULL,
+      dataMapping = NULL,
+      binWidth = NULL,
+      bins = NULL,
+      verticalLineFunctions = NULL,
+      verticalLineFunctionNames = NULL,
+      ...) {
+
+      super$initialize(...,
+                       title = title,
+                       subtitle = subtitle,
+                       xlabel = xlabel,
+                       ylabel = ylabel,
+                       watermark = watermark,
+                       data = data,
+                       metaData = metaData,
+                       dataMapping = dataMapping
+      ) #self$verticalLineProperties <- verticalLineProperties
+
+      self$binWidth = binWidth
+      self$bins = bins
+
+      self$verticalLineFunctions = verticalLineFunctions
+      #print( self$verticalLineFunctions )
+      self$verticalLineFunctionNames = verticalLineFunctionNames
+
+    },
+
+
+    addHistograms = function(plotObject,
+                             data,
+                             metaData = NULL,
+                             dataMapping ,
+                             binWidth = NULL , bins = NULL) {
+
+      binWidth <- ifnotnull(binWidth,binWidth,self$binWidth)
+      bins <- ifnotnull(bins,bins,self$bins)
+
+      self$mapData <- dataMapping$getMapData(data, metaData)
+
+      fill <- self$legend$titles$fill %||% "none"
+
+      # Define dummy unique value for grouping
+      # Allows further modification of the aesthetic property
+      if (is.null(self$legend$titles$fill)) {
+        self$mapData[, fill] <- as.factor(1)
+      }
+
+      plotObject <- plotObject + ggplot2::geom_histogram(
+        mapping = aes(x = self$mapData$x, fill = self$mapData[ , fill ]   ),
+        show.legend = TRUE,
+        binwidth = binWidth,
+        bins = bins,
+        alpha=.5
```

Comment:
alpha is a parameter defining the transparency of shaded areas
The name is commonly used by both Matlab and R
I think the goal is to prevent one histogram to hide another
I will be adding default sizes and alphas in Theme